whylog/regexassistant/__init__.py

```diff
@@ -0,0 +1,102 @@
+class RegexAssistant(object):
+    """
+    Responsible for creating regex for each line of Rule.
+    RegexAssistant helps user to write regex, it can also proposes regex.
+    One RegexAssistant per one entering Rule.
+    """
+
+    def _obvious_regex(self, text):
+        """
+        Creates regex form text by simple transformation:
+        - add backslash to special characters
+        - text -> '^' + text + '$'
+        :param text:
+        :return:
+        """
+        # not finished yet
+        return '^' + text + '$'
+
+    def __init__(self, line_objects=tuple()):
+        self.regexes = dict(
+            [
+                (line, self._obvious_regex(line.line_content)) for line in line_objects
+            ]
+        )
+        self.history = dict([(line, []) for line in line_objects])
+        pass
+
+    def add_lines(self, line_objects):
+        for line in line_objects:
+            self.regexes[line] = self._obvious_regex(line.line_content)
+            self.history[line] = []
+
+    def remove_lines(self, line_objects):
+        for line in line_objects:
+            del self.regexes[line]
+            del self.history[line]
```

Comment:
I can see here that `RegexAssistant` is responsible for keeping track of line history, adding data to history and removing it when needed. Perhaps the line object itself could keep track of this history? Then management would sort of manage by itself - one would just need to keep track of a collection of lines somewhere.